qiskit/synthesis/evolution/qdrift.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import SparsePauliOp, Pauli
+from qiskit.utils import algorithm_globals
+
+from .product_formula import ProductFormula
+from .lie_trotter import LieTrotter
+
+
+class QDrift(ProductFormula):
+    r"""The QDrift Trotterization method, which selects each each term in the
+    Trotterization randomly, with a probability proportional to its weight. Based on the work
+    of Earl Campbell in Ref. [1].
+
+    References:
+        [1]: E. Campbell, "A random compiler for fast Hamiltonian simulation" (2018).
+         `arXiv:quant-ph/1811.08017 <https://arxiv.org/abs/1811.08017>`_
+    """
+
+    def __init__(
+        self,
+        reps: int = 1,
+        insert_barriers: bool = False,
+        cx_structure: str = "chain",
+        atomic_evolution: Optional[
+            Callable[[Union[Pauli, SparsePauliOp], float], QuantumCircuit]
+        ] = None,
+    ) -> None:
+        r"""
+        Args:
+            reps: The number of times to repeat the Trotterization circuit.
+            insert_barriers: Whether to insert barriers between the atomic evolutions.
+            cx_structure: How to arrange the CX gates for the Pauli evolutions, can be
+                "chain", where next neighbor connections are used, or "fountain", where all
+                qubits are connected to one.
+            atomic_evolution: A function to construct the circuit for the evolution of single
+                Pauli string. Per default, a single Pauli evolution is decomopsed in a CX chain
+                and a single qubit Z rotation.
+        """
+        super().__init__(1, reps, insert_barriers, cx_structure, atomic_evolution)
+        self.sampled_ops = None
+
+    def synthesize(self, evolution):
+        # get operators and time to evolve
+        operators = evolution.operator
+        time = evolution.time
+
+        if not isinstance(operators, list):
+            pauli_list = [(Pauli(op), np.real(coeff)) for op, coeff in operators.to_list()]
+            coeffs = [np.real(coeff) for op, coeff in operators.to_list()]
+        else:
+            pauli_list = [(op, 1) for op in operators]
+            coeffs = [1 for op in operators]
+
+        # We artificially make the weights positive
+        weights = np.abs(coeffs)
+        lambd = np.sum(weights)
+
+        num_gates = 2 * (lambd ** 2) * (time ** 2)
+        factor = lambd * time / num_gates * self.reps
+        # The protocol calls for the removal of the individual coefficients,
+        # and multiplication by a constant factor.
+        scaled_ops = [(op, factor / coeff) for op, coeff in pauli_list]
+        self.sampled_ops = algorithm_globals.random.choice(
+            np.array(scaled_ops, dtype=object),
+            size=(int(np.ceil(num_gates * self.reps)),),
+            p=weights / lambd,
+        )
+
+        from qiskit.circuit.library.pauli_evolution import PauliEvolutionGate
```

Comment:
That hopefully fixes the cyclic import errors
```suggestion

        # pylint: disable=cyclic-import
        from qiskit.circuit.library.pauli_evolution import PauliEvolutionGate
```